liualgotrader/scanners/momentum.py

```diff
@@ -83,27 +85,27 @@ async def run_polygon(self) -> List[str]:
         tlog(f"{self.name}: run_polygon(): started")
         try:
             while True:
-                tickers = self.data_loader.data_api.get_symbols()
-                tlog(f"loaded {len(tickers)} tickers from Polygon")
-                if not len(tickers):
+                market_snapshot = self.data_loader.data_api.get_market_snapshot()
+                tlog(f"loaded {len(market_snapshot)} tickers of market snapshots from Polygon")
+                if not len(market_snapshot):
                     break
                 trade_able_symbols = await self._get_trade_able_symbols()
 
                 unsorted = [
-                    ticker
-                    for ticker in tickers
+                    _ticket_snapshot
+                    for _ticket_snapshot in market_snapshot
                     if (
-                        ticker["ticker"] in trade_able_symbols  # type: ignore
+                        _ticket_snapshot["ticker"] in trade_able_symbols  # type: ignore
```

Comment:
@riven314 I think it will be cleaner if that snapshot filtering will be implemented inside the data provider, using the specifics of the payload. I don't suppose all data providers will provide the same format of a payload . So instead of having scanner w/ run_polygon vs. run_alapca etc I'm thinking maybe we need to add some filtering criteria the get-snapshot functions, that will already return filtered data? or some other mechanism?

Reply:
this filtering rule is very specific to momentum strategy so I feel like its not the responsibility for data provider.
do you mean we should add a generic method for filtering symbols in alpaca & alpaca data provider? 
method like this could be used on other kind of filtering but it may add complexity on the implementation (e.g. what should be arguments for this method, how to pass in comparison operator ... etc.)

Reply:
I think what @amor71  is trying to imply is that the formatting of the payload should be within the data provider, that way things like momentum scanner can just call DataLoader.get_symbols and the data provider class for their selected provider will have consistent return. That way its easier for others to implement other providers like maybe IBKR or something. I think first we would need to define a class for the symbols so that we can standardise what information should be returned from a function list get_symbols. Then filtering would be in the scanner or strategy but because all data providers return the same it would eliminate the need for if blocks separating code dependent on the provider.